AdblockPlusSafari/AdblockPlusExtras.m

```diff
@@ -407,4 +389,33 @@ - (NSString *)filterListNameForTaskTaskIdentifier:(NSUInteger)taskIdentifier
   return nil;
 }
 
+- (void)onApplicationWillEnterForegroundNotification:(NSNotification *)notification
+{
+  if (self.reloading) {
+    return;
+  }
+
+  __weak __typeof(self) wSelf = self;
+  NSDate *lastActivity = wSelf.lastActivity;
+  wSelf.performingActivityTest = YES;
+  [SFContentBlockerManager reloadContentBlockerWithIdentifier:self.contentBlockerIdentifier completionHandler:^(NSError *error) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      if (error) {
+        NSLog(@"%@", error);
+      }
+      wSelf.performingActivityTest = NO;
```

Comment:
This almost the same code already exists in reloadWithCompletion, and it also drops the completion event (does not forward the callback anywhere). It only differs on outside - that this one is performing the hack activity test while the other does not. I see a chance for making the code cleaner by actually using the completion handler.